### DISCUSSION TITLE
Broadcasting a variable to a different domain - Positive electrode conductivity

### DISCUSSION THREAD

**Discussion by veera-visw** [pybamm-team/PyBaMM#4032] [Q&A]
I am simulating a aqueous alkaline battery (Nickel - MH) and I wanted to include an equation for the postivie electrode conductivity which varies as a function of cnorm : (c_s_p/c_max_p); where c_s_p is the surface concentration and c_max_p is max positive particel concentration.

A function parameter for sigma_p (positive electrode conductivity) as a function of cnorm causes a problem when I process the model due to a domain mismatch.

**children must have same or empty domains, not ['positive particle'] and ['positive electrode']**

Is there a way I can calculate the cnorm and assign it to a variable in the positive electrode domain to avoid this error?

Hope the description was clear. Thanks

**Comment by valentinsulzer** [comment]
How are you defining domains when you create c_s_p and c_max_p?

**Reply by veera-visw** [reply]
Something like this: 

c_s_p = pybamm.Variable(
    "Positive particle concentration [mol.m-3]",
    domain="positive particle",
    auxiliary_domains={
        "secondary": "positive electrode",
    },
)

c_max_p is a single floating number so its not assigmed to any domain.

**Comment by veera-visw** [comment]
Any suggestions?

**Comment by veera-visw** [comment]
Here is a possible solution that seems to work for me: Perhaps there is a better way to do it, but here is waht I am doing:

ct=pybamm.x_average(pybamm.surf(c_s_n))
ct2 = pybamm.PrimaryBroadcast(ct, "negative electrode")

Now, ct2 can be used to define a functionParameter for electrode conductivity 

inputs_sigma={"broadcasted particle conc":ct2}
sigma_n = pybamm.FunctionParameter("Negative electrode conductivity [S.m-1]", inputs_sigma)


This ofcourse means I cannot change the conductivity spatially, but this is a start.
